migration/alembic/versions/d1dc98f61aee_major_changes.py

```diff
@@ -0,0 +1,142 @@
+"""Major changes
+
+Revision ID: d1dc98f61aee
+Revises: 
+Create Date: 2024-07-29 11:42:19.723940
+
+"""
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+from alembic_postgresql_enum import TableReference
+from alembic_utils.pg_function import PGFunction
+from sqlalchemy import text as sql_text
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision: str = 'd1dc98f61aee'
+down_revision: Union[str, None] = None
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    sa.Enum('VALID', 'ON_HOLD', 'INVALID', name='stateenum').create(op.get_bind())
+    op.create_table('reference',
+        sa.Column('name', sa.String(), nullable=True),
+        sa.Column('alias', sa.String(), nullable=True),
+        sa.Column('assembly', sa.String(), nullable=True),
+        sa.Column('version', sa.String(), nullable=True),
+        sa.Column('taxon_id', sa.String(), nullable=True),
+        sa.Column('source', sa.String(), nullable=True),
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('deprecated', sa.Boolean(), nullable=False),
+        sa.Column('deleted', sa.Boolean(), nullable=False),
+        sa.Column('creation', sa.DateTime(timezone=True), server_default=sa.text('now()'), nullable=True),
+        sa.Column('modification', sa.DateTime(timezone=True), nullable=True),
+        sa.Column('extra_metadata', sa.JSON(), nullable=True),
+        sa.Column('ext_id', sa.Integer(), nullable=True),
+        sa.Column('ext_src', sa.String(), nullable=True),
+        sa.PrimaryKeyConstraint('id')
+    )
+    # Rename table patient to specimen
```

Comment:
@paulstretenowich I see below that the fms_id are going to be replaced by a combo of ext_id with ext_src in pretty much all tables.
Why doing it using 2 **add_column** operations and 1 **drop_column** instead of 1 **alter_column** and 1 **add_column**. Wouldn't it conserve the content of the column if any?

ext_src is currently a string but we should consider an enumerate ["freezeman", "migrated", "..."] to make it more robust. It could still be nullable.

Reply:
Good point, As so far in the database no fms_id is set it's fine but if it was the case it would need to be alter for the columns with an existing fms id.
I suggest we add that enum for ext_src in a second time as we don't have all values yet.